doc/samples/SqlConfigurableRetryLogic_StepByStep_CustomProvider.cs

```diff
@@ -0,0 +1,255 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+
+namespace CustomCRL_Doc
+{
+    class Program
+    {
+        private const string CnnStringFormat = "Server=localhost; Initial Catalog=Northwind__; Integrated Security=true; pooling=false; Timeout=1";
+
+        static void Main(string[] args)
+        {
+            AppContext.SetSwitch("Switch.Microsoft.Data.SqlClient.EnableRetryLogic", true);
+            RetryConnection(CnnStringFormat);
+        }
+
+        private static void RetryConnection(string connectionString)
+        {
+            // <Snippet1>
+            // Define the retry logic parameters
+            var options = new SqlRetryLogicOption()
+            {
+                // Tries 5 times before throwing an exception
+                NumberOfTries = 5,
+                // Prefered gap time to delay before retry
+                DeltaTime = TimeSpan.FromSeconds(1),
+                // Maximum gap time for each delay time before retry
+                MaxTimeInterval = TimeSpan.FromSeconds(20),
+                // SqlException retriable error numbers
+                TransientErrors = new int[] { 4060, 1024, 1025}
+            };
+            // </Snippet1>
+
+            // <Snippet2>
+            // Create a custom retry provider
+            SqlRetryLogicBaseProvider provider = CustomRetry.CreateCustomProvider(options);
+            // </Snippet2>
+
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                // <Snippet3>
+                // Assumes that connection is a valid SqlConnection object 
+                // Assign a retry provider to the connection
+                connection.RetryLogicProvider = provider;
+                // Establishing the connection will retry if one of the given transient failure or TimeoutException occurs.
```

Comment:
```suggestion
                // Establishing the connection will trigger retry if one of the given transient failure occurs.
```

I think you don't mean timeout here, plz check.